docs/content/en/http-methods.md

```diff
@@ -1,58 +1,277 @@
+**Examples:**
 
-</alert>
+```js
+const package = await $http.$get('https://unpkg.com/nuxt/package.json')
+```
 
-### `delete`
-### `get`
-### `head`
+```js
+// With prefixUrl option to call `https://example.com/items`
+const items = await $http.$get('items', { prefixUrl: 'https://example.com' })
+```
+
+### `$post`
+- arguments: `(url, body?, options?)`
+  - url: `String`
+  - body: `Object`
+  - options: [options](/options)
+- resolves: [JSON](https://developer.mozilla.org/en-US/docs/Web/API/Body/json) if [destr](https://github.com/nuxt-contrib/destr) can convert automatically, otherwise fallback to String
+- rejects: `Error`
+
+**Examples:**
+
+```js
+const data = await $http.$post('http://api.com', { foo: 'bar' })
+```
+
+```js
+// With some additional options
+const data = await $http.$post('http://api.com', { foo: 'bar' }, {
+  debug: true,
+  retry: 2,
+  serverTimeout: 5000
+})
+```
+
+### `$put`
+- arguments: `(url, body?, options?)`
+  - url: `String`
+  - body: `Object`
+  - options: [options](/options)
+- resolves: [JSON](https://developer.mozilla.org/en-US/docs/Web/API/Body/json) if [destr](https://github.com/nuxt-contrib/destr) can convert automatically, otherwise fallback to String
+- rejects: `Error`
+
+**Examples:**
+
+```js
+const data = await $http.$put('http://api.com/{id}', { foo: 'bar' })
+```
+
+```js
+// With some additional option
+const data = await $http.$put('http://api.com/{id}', { foo: 'bar' }, {
+  clientTimeout: 5000
+})
+```
 
+### `$delete`
 - arguments: `(url, options?)`
+  - url: `String`
+  - options: [options](/options)
+- resolves: [JSON](https://developer.mozilla.org/en-US/docs/Web/API/Body/json) if [destr](https://github.com/nuxt-contrib/destr) can convert automatically, otherwise fallback to String
+- rejects: `Error`
+
+**Example:**
+
+```js
+await $http.$delete('https://api.example.com/item/{id}')
+```
+
+```js
+// With some options to call `https://example.com/api/item`
+const jsonResponse = await $http.$delete('item/{id}', { 
+  baseUrl: 'https://example.com',
+  prefixUrl: '/api' 
+})
+```
+
+### `$patch`
+- arguments: `(url, body?, options?)`
+  - url: `String`
+  - body: `Object`
+  - options: [options](/options)
+- resolves: [JSON](https://developer.mozilla.org/en-US/docs/Web/API/Body/json) if [destr](https://github.com/nuxt-contrib/destr) can convert automatically, otherwise fallback to String
+- rejects: `Error`
+
+**Examples: **
+
+```js
+const data = await $http.$patch('http://api.com/{id}', { foo: 'bar' })
+```
+
+```js
+// With some additional option
+const data = await $http.$patch('http://api.com/{id}', { foo: 'bar' }, {
+  proxyHeaders: true,
+  proxyHeadersIgnore: ['content-type']
+})
+```
+
+### `$head`
+- arguments: `(url, options?)`
+  - url: `String`
+  - options: [options](/options)
+- resolves: [JSON](https://developer.mozilla.org/en-US/docs/Web/API/Body/json) if [destr](https://github.com/nuxt-contrib/destr) can convert automatically, otherwise fallback to String
```

Comment:
Head method doesn't return a response. It is basically for header testing

(`curl -v  -X HEAD https://unpkg.com/nuxt/package.json`)

Reply:
@pi0 i didn't see your note, thanks to remove it 👍